pandas/tseries/offsets.py

```diff
@@ -1160,6 +1161,92 @@ def onOffset(self, dt):
     _prefix = 'MS'
 
 
+class SemiMonthEnd(SingleConstructorOffset):
```

Comment:
make a common class which this inherits to put common routines (SemiMonthOffset)